.circleci/config.yml

```diff
@@ -0,0 +1,117 @@
+version: 2.1
+
+orbs:
+  codacy: codacy/base@1.0.0
+  codacy_plugins_test: codacy/plugins-test@0.4.0
+
+jobs:
+  dotnet:
+    parameters:
+      cmd:
+        type: string
+        description: "The command to run"
+      persist_to_workspace:
+        type: boolean
+        description: "Whether to persist the workspace or not at the end of the job"
+        default: false
+    docker:
+      #TODO: switch to docker image
+      - image: mcr.microsoft.com/dotnet/core/sdk:2.2
+    working_directory: ~/workdir
+    steps:
+      - run:
+          name: Install packages
+          command: |
+            echo "deb http://download.mono-project.com/repo/debian wheezy main" | tee /etc/apt/sources.list.d/mono-xamarin.list
+            apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+            apt-get update
+            apt-get install -y mono-complete build-essential nuget
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Run command - << parameters.cmd >>
+          command: << parameters.cmd >>
+      - when:
+          condition: << parameters.persist_to_workspace >>
+          steps:
+            - persist_to_workspace:
+                root: ~/
+                paths:
+                  - workdir/*
+  build_docker:
+    machine: true
+    working_directory: ~/workdir
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Publish tool docker locally
+          command: docker build . -t $CIRCLE_PROJECT_REPONAME:latest
+      - run:
+          name: Save docker to file
+          command: |
+            docker save --output docker-image.tar $CIRCLE_PROJECT_REPONAME:latest
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - workdir/docker-image.tar
+  publish:
+    machine: true
+    working_directory: ~/workdir
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Load docker from file
+          command: docker load --input docker-image.tar
+      - deploy:
+          name: Push application Docker image
+          command: |
+            docker login -u $DOCKER_USER -p $DOCKER_PASS
```

Comment:
just a note for the future, to refactor this thing in the orb

Reply:
this is a thing we need to do on all tools. I created an issue on codacy-orbs: https://github.com/codacy/codacy-orbs/issues/33